changelog/4121.breaking.rst

```diff
@@ -0,0 +1,3 @@
+`sunpy.map.GenericMap.top_right_coord` and `~sunpy.map.map.GenericMap.center`
```

Comment:
I think "bugfix" might be more appropriate for this changelog type, according to the changelog readme, a breaking change is:

> A change which requires users to change code and is not backwards compatible.

Which I don't think this really is, it's just a particularly bad bug fix lol